components/ftx/browser/ftx_json_parser.cc

```diff
@@ -0,0 +1,278 @@
+// Copyright (c) 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include <string>
+
+#include "brave/components/ftx/browser/ftx_json_parser.h"
+
+#include "base/json/json_reader.h"
+#include "base/logging.h"
+#include "base/strings/string_number_conversions.h"
+#include "base/values.h"
+
+bool FTXJSONParser::GetFuturesDataFromJSON(const std::string& json,
+                                           FTXFuturesData* data,
+                                           const std::string& filter) {
+  if (!data) {
+    return false;
+  }
+  base::JSONReader::ValueWithError value_with_error =
+      base::JSONReader::ReadAndReturnValueWithError(
+          json, base::JSONParserOptions::JSON_PARSE_RFC);
+  base::Optional<base::Value>& records_v = value_with_error.value;
+  if (!records_v) {
+    LOG(ERROR) << "Invalid response, could not parse JSON, JSON is: " << json;
```

Comment:
could this json potentially contain sensitive data or PII?

Reply:
Some of them not, some of them yes. I'll remove the JSON output for all of them. Thanks, I didn't notice these.

Reply:
These are all removed now